### PR TITLE
DEM/nfhl skid

### PR DESCRIPTION
**Pull request by jacobdadams**
This is the nfhl skid running in Cloud Run. It pulls data from FEMA's rest endpoints and uploads them to AGOL layers. 

It uses Cloud Run instead of Functions because of the time involved (a full run is >1 hr, assuming no layers time out). It's currently set to 2Gi RAM, and the last run used 97.93% of that. If one of the layers errors out, it's conceivable the retry will push it over the limit, so maybe 3Gi makes more sense. Or a Mi value inbetween.  

This should incorporate the latest dependabot CI group updates; the others are a moving target as the arcgis 2.2 update triggered a lot more changes, so we'll see how it runs after rebasing to main.

### CODE REVIEW

**Comment by steveoh** [comment]
> assuming no layers time out

This seems like enough reason to split the layers into separate jobs to me. Is a partial update better than no update?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM python:3.11-slim
```
Comment (steveoh) [inline Dockerfile:1]: good choice on slim. Some extra reading about [it vs alpine](https://www.reddit.com/r/docker/comments/g5hb93/alpine_vs_pythonslim_for_deploying_python_data/) if you're curious.

File: src/nfhl/config.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import logging
+import socket
+
+SKID_NAME = "nfhl_skid"
+
+AGOL_ORG = "https://utah-em.maps.arcgis.com"
+SENDGRID_SETTINGS = {  #: Settings for SendGridHandler
+    "from_address": "noreply@utah.gov",
+    "to_addresses": "jdadams@utah.gov",
```
Comment (steveoh) [inline src/nfhl/config.py:13]: **question**: how long do we expect this skid to live? If it's some what permanent, I recommend updating this to ugrc-developers.

**Review by steveoh — APPROVED** [review]

**Review by stdavis — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (jacobdadams) [reply Dockerfile:1]: Yeah, the problems building wheels is why I went with slim.

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+  pull_request:
+    branches:
+      - main
+      - dev
```
[the remark this answers is not in the document]
Comment (jacobdadams) [reply .github/workflows/push.yml:11]: Only whatever reason you guys had when I cribbed the original descendant of this file back in the first cloud function project I did.

**Comment by jacobdadams** [comment]
> This seems like enough reason to split the layers into separate jobs to me. Is a partial update better than no update?

It's already engineered to move on to the next if one layer fails. I am just worried that the retries may leave some residual memory usage that the garbage collector doesn't clean up, but this weekend's run had multiple fails/retries on the largest layers and didn't bomb out, so I think we're good.

**Pull request merged**